platform/ios/src/MGLMapView.mm

```diff
@@ -4759,7 +4804,11 @@ - (void)cameraIsChanging {
     }
 
     [self updateCompass];
-
+    
+    if (!self.scaleBarView.hidden) {
+        [(MGLScaleBarView *)self.scaleBarView setMetersPerPoint:[self metersPerPointAtLatitude:self.centerCoordinate.latitude]];
```

Comment:
The scale bar remains faded out until you set `metersPerPoint` making it safe. I'd prefer not to set `metersPerPoint` if the view is hidden to avoid excessive layout code from being executed.